File: project-euler.java/src/main/java/euler/solutions/_24.java
package euler.solutions;

import euler.common.BaseSolution;
import euler.common.Result;
import gnu.trove.list.TIntList;
import gnu.trove.list.array.TIntArrayList;

public class _24 extends BaseSolution {

    private TIntList digits;

    @Override
    public Result solve() {
        digits = new TIntArrayList(new int[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 });
        int n   = 1_000_000;
        int fac = 362880; // factorial(9)
        String ret = "";

        for (int i = 9; i > 0; i--) {
            int ord = (n / fac);
            n %= fac;
            if (n == 0) {
                ord--;
                n = fac;
            }
            fac /= i;
            int digit = nthDigit(ord);
            ret += digit;
        }

        ret += digits.get(0);

        return new Result(ret);
    }


    private int nthDigit(int idx)
    {
        int digit = digits.get(idx);
        digits.removeAt(idx);
        return digit;
    }
}

File: project-euler.go/solutions/_24/24.go
package _24

import "strconv"

var digits []int

func Solve() int64 {
	digits = []int{ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 }
	n   := 1000000
	fac := 362880 // factorial(9)
	ret := ""

	for i := 9; i > 0; i-- {
		ord := (n / fac)
		n %= fac
		if n == 0 {
			ord--
			n = fac
		}
		fac /= i
		digit := nthDigit(ord)
		ret += strconv.FormatInt(int64(digit), 10)
	}

	ret += strconv.FormatInt(int64(digits[0]), 10)

	retInt,_ := strconv.ParseInt(ret, 10, 64)
	return retInt
}

func nthDigit(idx int) int {
	digit := digits[idx]
	digits = append(digits[:idx], digits[idx+1:]...)
	return digit
}
